internal/services/appservice/function_app_function_resource.go

```diff
@@ -0,0 +1,490 @@
+			Default:     true,
+			Description: "Should this function be enabled. Defaults to `true`.",
+		},
+
+		"language": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty, // TODO - find the valida list of strings
+			Description:  "The language the Function is written in.",
+		},
+
+		"file": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MinItems: 1,
+			ForceNew: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+						Description:  "The filename of the file to be uploaded.",
+					},
+					"content": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+						Description:  "The content of the file.",
+					},
+				},
+			},
+		},
+
+		"test_data": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty, // Can an empty string be valid here?
+			Description:  "The test data for the function.",
+		},
+	}
+}
+
+func (r FunctionAppFunctionResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"config_url": {
+			Type:        pluginsdk.TypeString,
+			Computed:    true,
+			Description: "The URL of the configuration JSON.",
+		},
+
+		"url": {
+			Type:        pluginsdk.TypeString,
+			Computed:    true,
+			Description: "The function URL.",
+		},
+
+		"invocation_url": {
+			Type:        pluginsdk.TypeString,
+			Computed:    true,
+			Description: "The invocation URL.",
+		},
+
+		"script_url": {
+			Type:        pluginsdk.TypeString,
+			Computed:    true,
+			Description: "The script URL.",
+		},
+
+		"script_root_path_url": {
+			Type:        pluginsdk.TypeString,
+			Computed:    true,
+			Description: "The Script root path URL.",
+		},
+
+		"test_data_url": {
+			Type:        pluginsdk.TypeString,
+			Computed:    true,
+			Description: "The Test data URL.",
+		},
+
+		"secrets_file_url": {
+			Type:        pluginsdk.TypeString,
+			Computed:    true,
+			Description: "The URL for the Secrets File.",
+		},
+	}
+}
+
+func (r FunctionAppFunctionResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AppService.WebAppsClient
+
+			appFunction := FunctionAppFunctionModel{}
+			if err := metadata.Decode(&appFunction); err != nil {
+				return err
+			}
+
+			appId, err := parse.FunctionAppID(appFunction.AppID)
+			if err != nil {
+				return err
+			}
+
+			id := parse.NewFunctionAppFunctionID(appId.SubscriptionId, appId.ResourceGroup, appId.SiteName, appFunction.Name)
+
+			// TODO - Do we need to (or can we?) check the function runtime is ready?
```

Comment:
Another todo to check?

Reply:
one I forgot to remove in the latest commit 🙈